src/varity/vcf_lift.clj

```diff
@@ -0,0 +1,128 @@
+                        count))
+             (apply min))
+        new-alt
+        (map #(drop common-seq-length %) alt)
+        new-ref (drop common-seq-length ref)]
+    (cond
+      (and (zero? common-seq-length)
+           (not (some empty? new-alt))
+           (seq new-ref))
+      {:new-start (inc start) :new-end (inc end)
+       :new-alt (map #(apply str %) new-alt)
+       :new-ref (apply str new-ref)}
+
+      (or (some empty? new-alt)
+          (empty? new-ref))
+      (let [seq-pos (if (pos? start) start (dec end))
+            ref-seq (io-seq/read-sequence seq-reader
+                                          {:chr chr
+                                           :start seq-pos
+                                           :end seq-pos}
+                                          {:mask? false})]
+
+        (shift-interval chr (dec start) (- end common-seq-length)
+                        (concat new-ref ref-seq)
+                        (map #(concat % ref-seq) new-alt)
+                        seq-reader))
+
+      :else (shift-interval chr start (- end common-seq-length)
+                            new-ref new-alt seq-reader))))
+
+(defn- variant-compare [variant1 variant2]
+  (- (:pos variant1) (:pos variant2)))
+
+(defn liftover-variant
+  "Lifts over variants data from chains applied chain/index and seq-reader"
+  [seq-reader indexed-chains {:keys [chr pos ref alt info] :as variant}]
+  (when-let [{:keys [start end strand]}
+             (calc-new-interval chr pos
+                                (dec (+ pos (count ref))) indexed-chains)]
+    (let [{:keys [new-start new-end new-ref new-alt]}
+          (if (= strand :reverse)
+            (shift-interval (:chr variant) start end
+                            (map complement-table ref)
+                            (map #(map complement-table %) alt)
+                            seq-reader)
+            {:new-start start :new-end end
+             :new-ref ref
+             :new-alt alt})
+          check-ref (io-seq/read-sequence seq-reader
+                                          {:chr chr
+                                           :start new-start
+                                           :end new-end}
+                                          {:mark? false})
+          directed-ref (if (= strand :reverse)
+                         (apply str (reverse new-ref))
+                         new-ref)
+          directed-alt (if (= strand :reverse)
+                         (map #(apply str (reverse %)) new-alt)
+                         new-alt)]
+      (when (= directed-ref check-ref)
+        (->> (assoc variant
+                    :pos new-start
+                    :info (if (= (count alt) 1)
+                            (assoc info
+                                   :END new-end)
+                            info)
+                    :ref directed-ref
+                    :alt directed-alt)
+             (norm/realign seq-reader))))))
```

Comment:
`shift-interval` seems to work so similarly to `norm/realign`. Wouldn't it be enough just to realign once? 

```clojure
(defn liftover-variant
  [rdr chain {:keys [chr pos ref info] :as variant}]
  (when-let [{:keys [chr start end strand]} (calc-new-interval
                                             chr pos (dec (+ pos (count ref)))
                                             chain)]
    (let [ref-seq (io-seq/read-sequence rdr {:chr chr, :start start, :end end})]
      (when (= ref-seq (cond-> ref (= strand :reverse) util-seq/revcomp))
        (cond-> (assoc variant :chr chr :pos start)
          (:END info) (assoc-in [:info :END] (dec (+ start (count ref))))
          (= strand :reverse) (-> (update :ref util-seq/revcomp)
                                  (update :alt #(map util-seq/revcomp %)))
          true (#(norm/realign rdr %)))))))
```